### PR TITLE
Added AdThrive ad support to amp-ad

### PR DESCRIPTION
**Pull request by jamarzka**
Added [AdThrive](http://www.adthrive.com) ad support to amp-ad.

### CODE REVIEW

**Comment by jamarzka** [comment]
CLA is signed as a corporation: CafeMedia

File: ads/adthrive.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+distributed under the License is distributed on an "AS-IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# AdThrive
+
+## Examples
+
```
Comment (zhouyx) [inline ads/adthrive.md]: I would suggest adding description to each example for clearance.

File: ads/adthrive.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+import {loadScript, validateData} from '../3p/3p';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function adthrive(global, data) {
+  validateData(data, ['siteId', 'adUnit'], ['sizes']);
+  loadScript(global, 'https://ads.adthrive.com/sites/' + data.siteId + '/amp.min.js');
```
Comment (zhouyx) [inline ads/adthrive.js]: `encodeURIComponent(data.siteId)`

**Comment by zhouyx** [comment]
@jamarzka LGTM in general. To fix travis, please add your declared data object variables to ads.extern.js

**Comment by zhouyx** [comment]
@jamarzka Also I am not able to find CafeMedia in the CLA list. When did you sign it?

File: ads/adthrive.md (hunk 1 above)
Comment (jamarzka) [inline ads/adthrive.md]: @zhouyx I added some simple descriptions

File: ads/adthrive.js (hunk 2 above)
Comment (jamarzka) [inline ads/adthrive.js]: @zhouyx done

**Comment by jamarzka** [comment]
@zhouyx I added the only data variable that is used to ads.extern.js but the build is still failing. Anything else I need to do?

File: ads/adthrive.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      data-ad-unit="AdThrive_Content_1">
+  </amp-ad>
+```
+
+### Render an ad with a fixed size 320x50
+```html
+  <amp-ad width="320" height="50"
+      type="adthrive"
+      data-site-id="test"
+      data-ad-unit="AdThrive_Content_1",
```
Comment (zhouyx) [inline ads/adthrive.md]: Ha! This is why travis is failing. Please remove `,`

File: ads/adthrive.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      data-sizes="320x50">
+  </amp-ad>
+```
+
+### Render an ad with multiple sizes (320x50,320x100,300x250)
+```html
+  <amp-ad width="320" height="50"
+      type="adthrive"
+      data-site-id="test"
+      data-ad-unit="AdThrive_Content_1",
```
Comment (zhouyx) [inline ads/adthrive.md]: same

File: examples/ads.amp.html [hunk 5]
```diff
@@ -320,6 +321,26 @@ <h2 id="adtech">AdTech (1x1 fake image ad)</h2>
     <div fallback></div>
   </amp-ad>
 
+  <h2 id="adthrive">AdThrive 320x50 banner</h2>
+  <amp-ad width="320" height="50"
+      type="adthrive"
+      data-site-id="test"
+      data-ad-unit="AdThrive_Content_1",
```
Comment (zhouyx) [inline examples/ads.amp.html]: here as well

File: examples/ads.amp.html [hunk 6]
```diff
@@ -320,6 +321,26 @@ <h2 id="adtech">AdTech (1x1 fake image ad)</h2>
...
+      data-sizes="320x50">
+    <div placeholder></div>
+    <div fallback></div>
+  </amp-ad>
+
+  <h2 id="adthrive">AdThrive 300x250 banner</h2>
+  <amp-ad width="300" height="250"
+      type="adthrive"
+      data-site-id="test"
+      data-ad-unit="AdThrive_Content_2",
```
Comment (zhouyx) [inline examples/ads.amp.html]: same

File: ads/adthrive.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+import {loadScript, validateData} from '../3p/3p';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function adthrive(global, data) {
+  validateData(data, ['siteId', 'adUnit'], ['sizes']);
+  loadScript(global, 'https://ads.adthrive.com/sites/' + encodeURIComponent(data.siteId) + '/amp.min.js');
```
Comment (zhouyx) [inline ads/adthrive.js]: please 80 chars max each line. Also extra 2 indents for the second line.

File: ads/adthrive.md (hunk 4 above)
Comment (zhouyx) [inline ads/adthrive.md]: same

**Comment by jamarzka** [comment]
@zhouyx I completed the requested changes and it looks like the build was successful now.

File: ads/adthrive.js [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+import {loadScript, validateData} from '../3p/3p';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function adthrive(global, data) {
+  validateData(data, ['siteId', 'adUnit'], ['sizes']);
+  loadScript(global, 'https://ads.adthrive.com/sites/'
+      + encodeURIComponent(data.siteId) + '/amp.min.js');
```
Comment (zhouyx) [inline ads/adthrive.js]: oops, i might not be quite clear. but 4 indents here please.

Comment (jamarzka) [inline ads/adthrive.js]: Do you mean 8 spaces then? Is there a linting test I'm missing that checks for this? I based my indentation on how these scripts handled it: adreactor.js, advertserve.js, and brainy.js.

Comment (zhouyx) [inline ads/adthrive.js]: Ah! Sorry, fail to read the `+ `. This looks good

**Comment by zhouyx** [comment]
@jamarzka Thanks for the Pull Request. Merged

**Pull request merged**